// @typeschema/valibot@0.13.5/dist/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  assert: () => assert,
  serializationAdapter: () => serializationAdapter,
  toJSONSchema: () => toJSONSchema,
  validate: () => validate,
  validationAdapter: () => validationAdapter,
  wrap: () => wrap
});
module.exports = __toCommonJS(src_exports);
var import_core3 = require("@typeschema/core");

// src/serialization.ts
var import_core = require("@typeschema/core");
var importSerializationModule = (0, import_core.memoize)(async () => {
  try {
    var dynamicallyImportedModule = await import(
      /* webpackIgnore: true */
      "@gcornut/valibot-json-schema"
    );
  } catch (moduleImportError) {
    throw moduleImportError;
  }
  const { toJSONSchema: toJSONSchema2 } = dynamicallyImportedModule;
  return { toJSONSchema: toJSONSchema2 };
});
var serializationAdapter = async (schema) => {
  const { toJSONSchema: toJSONSchema2 } = await importSerializationModule();
  return toJSONSchema2({ ignoreUnknownValidation: true, schema });
};

// src/validation.ts
var import_core2 = require("@typeschema/core");
var importValidationModule = (0, import_core2.memoize)(async () => {
  try {
    var dynamicallyImportedModule = await import(
      /* webpackIgnore: true */
      "valibot"
    );
  } catch (moduleImportError) {
    throw moduleImportError;
  }
  const { getDotPath, safeParseAsync } = dynamicallyImportedModule;
  return { getDotPath, safeParseAsync };
});
var validationAdapter = async (schema) => {
  const { getDotPath, safeParseAsync } = await importValidationModule();
  return async (data) => {
    const result = await safeParseAsync(schema, data);
    if (result.success) {
      return {
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        data: result.output,
        success: true
      };
    }
    return {
      issues: result.issues.map((issue) => ({
        message: issue.message,
        path: getDotPath(issue)?.split(".")
      })),
      success: false
    };
  };
};

// src/index.ts
var validate = (0, import_core3.createValidate)(validationAdapter);
var assert = (0, import_core3.createAssert)(validate);
var wrap = (0, import_core3.createWrap)(assert, validate);
var toJSONSchema = (0, import_core3.createToJSONSchema)(serializationAdapter);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  assert,
  serializationAdapter,
  toJSONSchema,
  validate,
  validationAdapter,
  wrap
});
